### PR TITLE
[CI/Build] Fix structural tag test for xgrammar 0.2.3 (`any_order` field)

### PR DESCRIPTION
**Pull request by khluu**
## Summary

The CPU CI job **Async Engine, Inputs, Utils, Worker, Config (CPU)** started failing on `tests/tool_parsers/test_structural_tag_registry.py::test_get_model_structural_tag_supports_vllm_hermes`:

```
assert tag.model_dump() == {...}
...
E       +   'any_order': False,
```

### Root cause

[xgrammar 0.2.3](https://pypi.org/project/xgrammar/0.2.3/) (released 2026-06-27, [mlc-ai/xgrammar#668](https://github.com/mlc-ai/xgrammar/pull/668) *"support any_order in json_schema format"*) added an `any_order` field to the json_schema structural-tag format. As a result `StructuralTag.model_dump()` now serializes `"any_order": False` inside each tag's `content` block, which the test's hardcoded expected dict did not contain.

`requirements/common.txt` pinned xgrammar with an open range (`>= 0.2.1, < 1.0.0`). The CPU image builds fresh each run and re-resolves that range, so it picked up the brand-new 0.2.3 two days after release. The test runs **only** in the CPU lane — `tests/tool_parsers/` is CPU-only (no GPU job executes it) — which is why no other job went red.

### Fix (adopt 0.2.3 rather than pin below it)

- Bump the floor to `xgrammar >= 0.2.3` in `requirements/common.txt`.
- Sync the ROCm test lockfile pin to `0.2.3` (kept consistent; `pip-compile-rocm` pre-commit hook passes).
- Add `"any_order": False` to the two expected `content` dicts in the test.

Pinning `< 0.2.3` instead would only defer the same test update to the next bump, so we move forward onto the maintained line.

## Test

- `pre-commit run --all-files` equivalent ran on commit — all hooks pass (`ruff`, `mypy`, `pip-compile`, `pip-compile-rocm`).
- Verified against a clean install of `xgrammar==0.2.3` that `JSONSchemaFormat.model_dump()` emits `"any_order": false` as a sibling of `"style"`, exactly matching the updated expectations:
  ```
  {"type": "json_schema", "json_schema": {...}, "style": "json", "any_order": false}
  ```
  The full vLLM test could not be run locally (no GPU/precompiled env in this checkout); it should be confirmed green by the CPU CI lane on this PR.

## Notes

- **Not a duplicate**: no open PR addresses the xgrammar 0.2.3 `any_order` serialization break (checked open PRs matching `xgrammar`, `any_order`, `structural tag`).
- AI assistance (Claude) was used for this change.

### CODE REVIEW

**Comment by sfeng33** [comment]
Superseded by https://github.com/vllm-project/vllm/pull/47015

**Comment by khluu** [comment]
## 🧪 Test-selection report — PR #47068

**Changed files:** 3 · **labels:** _(none)_
**Old (`source_file_dependencies`):** 151 jobs → **New (selector):** 151 jobs (of 207)
**Verdict:** ✅ **looks reasonable** — every changed path has a specific label; no high-risk drops

<details><summary>🔻 Dropped vs old — 0 (old runs, selector skips)</summary>

_none_

</details>

<details><summary>🔺 Added vs old — 0 (selector runs, old skips)</summary>

_none_

</details>

<details><summary>▶️ Selector-selected — 151</summary>

- `[Abuild]` :docker: :smoking: Non-root smoke tests — `image_build`
- `[Abuild]` :docker: Build CPU image — `image_build`
- `[Abuild]` :docker: Build HPU image — `image_build`
- `[Abuild]` :docker: Build arm64 image — `runall_pattern`
- `[Abuild]` :docker: Build image — `image_build`
- `[Attention]` V1 attention (B200) — `runall_pattern`
- `[Attention]` V1 attention (H100-MI300) — `runall_pattern`
- `[Basic Correctness]` Basic Correctness — `runall_pattern`
- `[Benchmarks]` Benchmarks CLI Test — `runall_pattern`
- `[CPU]` Arm CPU Test — `runall_pattern`
- `[CPU]` CPU-Distributed Tests (DP+TP) — `runall_pattern`
- `[CPU]` CPU-Distributed Tests (PP+TP) — `runall_pattern`
- `[CPU]` CPU-Kernel Tests — `runall_pattern`
- `[CPU]` CPU-Language Generation and Pooling Model Tests — `runall_pattern`
- `[CPU]` CPU-ModelRunnerV2 Tests — `runall_pattern`
- `[CPU]` CPU-Multi-Modal Model Tests %N — `runall_pattern`
- `[CPU]` CPU-Quantization Model Tests — `runall_pattern`
- `[CUDA]` Cudagraph — `runall_pattern`
- `[CUDA]` Platform Tests (CUDA) — `runall_pattern`
- `[Compile]` Distributed Compile Unit Tests (2xH100) — `runall_pattern`
- `[Compile]` Fusion E2E Config Sweep (H100) — `runall_pattern`
- `[Compile]` Fusion E2E Quick (H100) — `runall_pattern`
- `[Compile]` Fusion E2E TP2 (B200) — `runall_pattern`
- `[Compile]` Fusion E2E TP2 AR-RMS Config Sweep (H100) — `runall_pattern`
- `[Compile]` Fusion E2E TP2 AsyncTP Config Sweep (H100) — `runall_pattern`
- `[Compile]` Fusion E2E TP2 Quick (H100) — `runall_pattern`
- `[Compile]` Fusion and Compile Unit Tests (2xB200) — `runall_pattern`
- `[Compile]` Sequence Parallel Correctness Tests (2 GPUs) — `runall_pattern`
- `[Disaggregated]` CrossLayer KV layout Distributed NixlConnector PD accuracy tests (4 GPUs) — `runall_pattern`
- `[Disaggregated]` DP EP Distributed NixlConnector PD accuracy tests (4 GPUs) — `runall_pattern`
- `[Disaggregated]` Distributed FlashInfer NixlConnector PD accuracy (4 GPUs) — `runall_pattern`
- `[Disaggregated]` Distributed NixlConnector PD accuracy (4 GPUs) — `runall_pattern`
- `[Disaggregated]` Hybrid SSM NixlConnector PD accuracy tests (4 GPUs) — `runall_pattern`
- `[Disaggregated]` Hybrid SSM NixlConnector PD prefix cache test (2 GPUs) — `runall_pattern`
- `[Disaggregated]` MultiConnector (Nixl+Offloading) PD accuracy (2 GPUs) — `runall_pattern`
- `[Disaggregated]` MultiConnector (Nixl+Offloading) PD edge cases (2 GPUs) — `runall_pattern`
- `[Disaggregated]` NixlConnector PD + Spec Decode acceptance (2 GPUs) — `runall_pattern`
- `[Distributed]` Distributed Comm Ops — `runall_pattern`
- `[Distributed]` Distributed Compile + Comm (4 GPUs) — `runall_pattern`
- `[Distributed]` Distributed Compile + RPC Tests (2 GPUs) — `runall_pattern`
- `[Distributed]` Distributed DP Tests (2 GPUs) — `runall_pattern`
- `[Distributed]` Distributed DP Tests (4 GPUs) — `runall_pattern`
- `[Distributed]` Distributed Tests (8 GPUs)(H100) — `runall_pattern`
- `[Distributed]` Distributed Torchrun + Examples (4 GPUs) — `runall_pattern`
- `[Distributed]` Distributed Torchrun + Shutdown Tests (2 GPUs) — `runall_pattern`
- `[Distributed]` Pipeline + Context Parallelism (4 GPUs) — `runall_pattern`
- `[Distributed]` RayExecutorV2 (4 GPUs) — `runall_pattern`
- `[Docker]` Docker Build Metadata — `runall_pattern`
- `[Engine]` Engine — `runall_pattern`
- `[Engine]` Engine (1 GPU) — `runall_pattern`
- `[Engine]` e2e Core (1 GPU) — `runall_pattern`
- `[Engine]` e2e Scheduling (1 GPU) — `runall_pattern`
- `[Entrypoints]` Entrypoints Integration (API Server Generate) — `runall_pattern`
- `[Entrypoints]` Entrypoints Integration (API Server OpenAI - Part 1) — `runall_pattern`
- `[Entrypoints]` Entrypoints Integration (API Server OpenAI - Part 2) — `runall_pattern`
- `[Entrypoints]` Entrypoints Integration (API Server) — `runall_pattern`
- `[Entrypoints]` Entrypoints Integration (LLM) — `runall_pattern`
- `[Entrypoints]` Entrypoints Integration (Multimodal) — `runall_pattern`
- `[Entrypoints]` Entrypoints Integration (Pooling) — `runall_pattern`
- `[Entrypoints]` Entrypoints Integration (Responses API) — `runall_pattern`
- `[Entrypoints]` Entrypoints Integration (Speech to Text) — `runall_pattern`
- `[Entrypoints]` Entrypoints Unit Tests — `runall_pattern`
- `[Entrypoints]` OpenAI API Correctness — `runall_pattern`
- `[Expert Parallelism]` EPLB Algorithm — `runall_pattern`
- `[Expert Parallelism]` EPLB Execution — `runall_pattern`
- `[Expert Parallelism]` Elastic EP Scaling Test — `runall_pattern`
- `[Hardware]` Ascend NPU Test — `runall_pattern`
- `[Hardware]` Intel HPU Test — `runall_pattern`
- `[Hardware - AMD Build]` AMD: :docker: build test image and artifacts — `image_build`
- `[Hardware - AMD Build]` AMD: :docker: ensure ci_base — `image_build`
- `[Intel]` :docker: Build XPU image — `image_build`
- `[Kernels]` Deepseek V4 Kernel Test (B200) — `runall_pattern`
- `[Kernels]` Deepseek V4 Kernel Test (H100) — `runall_pattern`
- `[Kernels]` Kernels (B200) — `runall_pattern`
- `[Kernels]` Kernels Attention DiffKV Test (H100) — `runall_pattern`
- `[Kernels]` Kernels Attention Test %N — `runall_pattern`
- `[Kernels]` Kernels Core Operation Test — `runall_pattern`
- `[Kernels]` Kernels DeepGEMM Test (H100) — `runall_pattern`
- `[Kernels]` Kernels FusedMoE Layer Test (2 B200s) — `runall_pattern`
- `[Kernels]` Kernels FusedMoE Layer Test (2 H100s) — `runall_pattern`
- `[Kernels]` Kernels Helion Test — `runall_pattern`
- `[Kernels]` Kernels KDA Test — `runall_pattern`
- `[Kernels]` Kernels Mamba Test — `runall_pattern`
- `[Kernels]` Kernels MiniMax Reduce RMS Test (2 GPUs) — `runall_pattern`
- `[Kernels]` Kernels MoE Test %N — `runall_pattern`
- `[Kernels]` Kernels Quantization Test %N — `runall_pattern`
- `[Kernels]` vLLM IR Tests — `runall_pattern`
- `[LM Eval]` LM Eval Small Models — `runall_pattern`
- `[LM Eval]` LM Eval TurboQuant KV Cache — `runall_pattern`
- `[LM Eval]` MRCR Eval Small Models — `runall_pattern`
- `[LoRA]` LoRA %N — `runall_pattern`
- `[LoRA]` LoRA TP (Distributed) — `runall_pattern`
- `[Miscellaneous]` Async Engine, Inputs, Utils, Worker — `runall_pattern`
- `[Miscellaneous]` Async Engine, Inputs, Utils, Worker, Config (CPU) — `test_cmd_owner`
- `[Miscellaneous]` Batch Invariance (A100) — `runall_pattern`
- `[Miscellaneous]` Batch Invariance (B200) — `runall_pattern`
- `[Miscellaneous]` Batch Invariance (H100) — `runall_pattern`
- `[Miscellaneous]` Examples — `runall_pattern`
- `[Miscellaneous]` Extract Hidden States Integration — `runall_pattern`
- `[Miscellaneous]` Extract Hidden States Integration (2 GPUs) — `runall_pattern`
- `[Miscellaneous]` Metrics, Tracing (2 GPUs) — `runall_pattern`
- `[Miscellaneous]` Regression — `runall_pattern`
- `[Miscellaneous]` V1 Core + KV + Metrics — `runall_pattern`
- `[Miscellaneous]` V1 Others (CPU) — `runall_pattern`
- `[Miscellaneous]` V1 Sample + Logits — `runall_pattern`
- `[Miscellaneous]` V1 Spec Decode — `runall_pattern`
- `[Model Executor]` Model Executor — `runall_pattern`
- `[Model Runner V2]` Model Runner V2 Core Tests — `runall_pattern`
- `[Model Runner V2]` Model Runner V2 Distributed (2 GPUs) — `runall_pattern`
- `[Model Runner V2]` Model Runner V2 Examples — `runall_pattern`
- `[Model Runner V2]` Model Runner V2 Pipeline Parallelism (4 GPUs) — `runall_pattern`
- `[Model Runner V2]` Model Runner V2 Spec Decode — `runall_pattern`
- `[Models - Basic]` Basic Models Test (Other CPU) — `runall_pattern`
- `[Models - Basic]` Basic Models Tests (Extra Initialization) %N — `runall_pattern`
- `[Models - Basic]` Basic Models Tests (Initialization) — `runall_pattern`
- `[Models - Basic]` Basic Models Tests (Other) — `runall_pattern`
- `[Models - Distributed]` Distributed Model Tests (2 GPUs) — `runall_pattern`
- `[Models - Language]` Language Models Tests (Extra Standard) %N — `runall_pattern`
- `[Models - Language]` Language Models Tests (Hybrid) %N — `runall_pattern`
- `[Models - Language]` Language Models Tests (Standard) — `runall_pattern`
- `[Models - Multimodal]` Multi-Modal Accuracy Eval (Small Models) — `runall_pattern`
- `[Models - Multimodal]` Multi-Modal Models (Standard) 1: qwen2 — `runall_pattern`
- `[Models - Multimodal]` Multi-Modal Models (Standard) 2: qwen3 + gemma — `runall_pattern`
- `[Models - Multimodal]` Multi-Modal Models (Standard) 3: llava + qwen2_vl — `runall_pattern`
- `[Models - Multimodal]` Multi-Modal Models (Standard) 4: other + whisper — `runall_pattern`
- `[Models - Multimodal]` Multi-Modal Processor — `runall_pattern`
- `[Models - Multimodal]` Multi-Modal Processor (CPU) — `runall_pattern`
- `[Plugins]` Plugin Tests (2 GPUs) — `runall_pattern`
- `[PyTorch]` PyTorch Compilation Passes Unit Tests — `runall_pattern`
- `[PyTorch]` PyTorch Compilation Unit Tests — `runall_pattern`
- `[PyTorch]` PyTorch Compilation Unit Tests (H100) — `runall_pattern`
- `[PyTorch]` PyTorch Fullgraph — `runall_pattern`
- `[PyTorch]` PyTorch Fullgraph Smoke Test — `runall_pattern`
- `[PyTorch]` Pytorch Nightly Dependency Override Check — `runall_pattern`
- `[Quantization]` Quantization — `runall_pattern`
- `[Quantization]` Quantized Fusions — `runall_pattern`
- `[Quantization]` Quantized MoE Test (B200) — `runall_pattern`
- `[Quantization]` Quantized Models Test — `runall_pattern`
- `[Ray Compatibility]` Ray Dependency Compatibility Check — `runall_pattern`
- `[Rust Frontend Cargo]` Rust Frontend Cargo Style + Clippy — `runall_pattern`
- `[Rust Frontend Cargo]` Rust Frontend Cargo Tests — `runall_pattern`
- `[Rust Frontend E2E]` Rust Frontend Core Correctness — `runall_pattern`
- `[Rust Frontend E2E]` Rust Frontend Distributed — `runall_pattern`
- `[Rust Frontend E2E]` Rust Frontend OpenAI Coverage — `runall_pattern`
- `[Rust Frontend E2E]` Rust Frontend Serve/Admin Coverage — `runall_pattern`
- `[Rust Frontend E2E]` Rust Frontend Tool Use — `runall_pattern`
- `[Samplers]` Samplers Test — `runall_pattern`
- `[Spec Decode]` Spec Decode Draft Model — `runall_pattern`
- `[Spec Decode]` Spec Decode Eagle — `runall_pattern`
- `[Spec Decode]` Spec Decode Ngram + Suffix — `runall_pattern`
- `[Spec Decode]` Spec Decode Speculators + MTP — `runall_pattern`

</details>

<sub>report-only (no gating) · selector `sha256:ba598ef5615e` · script `/Users/khluu/select_tests.py` · base `72f639927f6c` · head `2b30cdf9f428`</sub>
